gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreBranch.java

```diff
@@ -54,9 +54,10 @@ public String getName() {
 
   public abstract String getBranchTypeString(boolean capitalized);
 
+  @SuppressWarnings("regexp")
   @Override
-  public GitCoreCommit getPointedCommit() throws GitCoreException {
-    GitCoreCommit gitCoreCommit = pointedCommitRef.get();
+  public synchronized GitCoreCommit getPointedCommit() throws GitCoreException {
+    var gitCoreCommit = pointedCommitRef.get();
```

Comment:
Ditto, `AtomicReference` likely unneeded given that the method is synchronized

Reply:
will be resolved on #281